dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -553,7 +556,14 @@ boolean isUnusedClassSymbolReference(ClassSymbolReference reference) {
       if (targetConstantPoolIndices.isEmpty()) {
         // This reference is not found in sourceJavaClass any more. This means that there exist
         // overlapping classes in the class path and that the first one does not use target class.
-        return true;
+        logger.warning(
+            "The source "
+                + sourceClassName
+                + " does not have target "
+                + targetClassName
+                + ", because of overlapping source classes on the class path");
+        // By returning false ("the reference used"), the reference is shown as error
+        return false;
```

Comment:
I'm not sure this is right. Given this class path, this potential problem will never actually cause a runtime error, right?